cassiopeia/mixins/errors.py

```diff
@@ -77,3 +77,12 @@ class UnknownCigarStringError(Exception):
 
 class UnspecifiedConfigParameterError(Exception):
     pass
+
+
+class BranchLengthEstimatorError(Exception):
+    """An Exception class for the BranchLengthEstimator class."""
+    pass
+
+
+class IIDExponentialMLEError(BranchLengthEstimatorError):
```

Comment:
Why do we need a specific error for the IIDExpenentialMLE class? Is this best practice?

Reply:
It is common practice: subclassing exceptions allows one to be more specific when catching exceptions.

Reply:
Sounds good!